designs/admin-api-authentication.md

```diff
@@ -0,0 +1,69 @@
+---
+state: draft
+---
+
+# Admin API Authentication function
+
+## Summary
+The authentication function of Node-RED supports the 'credentials' scheme and the 'strategy' scheme, but there is a problem that the Admin API can not be used in the 'strategy' scheme .
+
+This design note proposes a feature that enables the execution of the Admin API even in the 'strategy' scheme by adding a customizable token-based authentication function.
+
+## Authors
+
+ - @KazuhiroIto
+
+## Details
+
+### Current behavior
+- `adminAuth` defines the authentication scheme used by the Admin API.
+- The authentication scheme results in a token being generated.
+- This token is then included in future http requests to authenticate the request.
+- This authentication is done by comparing the token to the list of known tokens that Node-RED maintains.
+- For the 'credentials' scheme, tokens can be generated by doing an HTTP POST to `/auth/token`.
+- For the "strategy" scheme, this is not possible. Because the OAuth strategy requires interaction with the browser, the Admin API cannot be executed.
+  
+### Adding a customizable token-based authentication function
+Add a new option `tokens` and `tokenHedaer` to `adminAuth`.
+
+The `tokens` options is a user-defined function process that has a 'token' as an argument, and can verify the provided token.
+This function is called if the token is not recognized from Node-RED's internal token store.
+
+The token is obtained from the existing Authorization header.
+If you want to get the token from any custom header, use the `tokenHedaer` option.
+You can obtain a token from the specified header name instead of the authorization header by specifying the header name in `tokenHedaer` option.
```

Comment:
```suggestion
You can obtain a token from the specified header name instead of the authorization header by specifying the header name in `tokenHeader` option.
```